server/athenian/api/controllers/features/entries.py

```diff
@@ -545,6 +550,122 @@ async def calc_developer_metrics_github(devs: Sequence[Collection[str]],
     return result, topics_seq
 
 
+async def get_calculator_for_user(
+    service: str,
```

Comment:
Marvin, the ideology of the code inside "features" and "miners" is that it is completely separate from the state. That is, if you look everywhere around, I don't accept `sdb`. Is it possible to re-architect this to choose which calculator to use _outside_ of `entries.py` - for example, in a controller, in a new module, or in `Settings` class (`settings.py`) :pray:  It will help to retain the (kind of) clean architecture separation of responsibilities.

Reply:
Sure! It makes sense, thanks for pointing it out 👍

Reply:
Done.

One thing that I was considering is that having all these funcs in `entries.py` as `classmethod`s of a class would make the overriding of the logic much easier. WDYT?

Otherwise how for example would you override the logic of the `PullRequestMiner`? Among the things that I considered:
- make these functions accept a `PullRequestMiner` parameter so that the experiment will just call the original function and will pass a `PullRequestMinerWhatever`, but it doesn't seem elegant to change the signature just for this,
- make everything `classmethod`s and the `get_selector` function instead of returning the calculator function will return the class variation that will have the responsibility also of calling the default implementation (this will be granted if the variation class inherits from the original class). This sounds much cleaner to me.
- hack it with monkey-patching or even with `unittest.mock`, but I really don't wanna go into this direction. 😄

Reply:
Yes, having a class is so much better. In fact, let's have those functions as regular methods instead of `classmethod`, and accept the common arguments like `pdb`, `mdb`, `rdb`, etc. in `__init__`. Caveat: you'll have some fun time updating the unit tests.

Reply:
> classmethod, and accept the common arguments like pdb, mdb, rdb, etc. in __init__. 

I like it, deal! 👍 

> Caveat: you'll have some fun time updating the unit tests.

😂

Reply:
OK, please ping when ready :+1: